frontend/src/components/pages/RegistrantExperience/Waiver/index.tsx

```diff
@@ -1,13 +1,175 @@
+
+        <OrderedList pl="10" spacing={5}>
+          {waiverInterface.requiredClauses.map(
+            (clause: RequiredClauseResponse, index: number) => (
+              <ListItem textStyle="bodyRegular" key={index}>
+                {clause.text}
+              </ListItem>
+            ),
+          )}
+        </OrderedList>
+      </VStack>
+
+      <VStack spacing={5} pt="10" align="stretch">
+        <Text textStyle="heading">
+          By clicking the checkbox below, you agree to the terms and conditions
+          above.
+        </Text>
+        <Checkbox
+          isChecked={waiverInterface.agreedRequiredClauses}
+          onChange={() =>
+            waiverDispatch({ type: WaiverActions.ClICK_REQUIRED_CLAUSE })
+          }
+        >
+          <Text textStyle="bodyRegular">
+            I agree to the above identity and liability waiver conditions{" "}
+            <Text as="span" color="text.critical.100" verticalAlign="super">
+              *
+            </Text>
+          </Text>
+        </Checkbox>
+      </VStack>
+      <Divider pt="4" pb="4" borderColor="border.secondary.100" />
+      <VStack spacing={5} pt="8" align="stretch">
+        <Text color="primary.green.100" textStyle="heading">
+          Additional Clauses
+        </Text>
+        {waiverInterface.optionalClauses.map(
+          (clause: OptionalClauseResponse, index: number) => (
+            <VStack spacing={6} pt="2" align="stretch" key={index}>
+              <Text textStyle="bodyRegular">{clause.text}</Text>
+              <RadioGroup
+                onChange={() =>
+                  waiverDispatch({
+                    type: WaiverActions.CLICK_OPTIONAL_CLAUSE,
+                    optionalClauseId: index,
+                  })
+                }
+                key={index}
+                value={String(clause.agreed)}
+              >
+                <Stack spacing={5} direction="row">
+                  <Radio value="true" mb="0" textStyle="bodyRegular">
+                    {/* NOTE!?: I use mb='0' because there's a margin misalignment issue with Radio component */}
+                    I agree
+                  </Radio>
+                  <Radio value="false" textStyle="bodyRegular">
+                    I{" "}
+                    <Text as="span" textStyle="bodyBold">
+                      do not
+                    </Text>{" "}
+                    agree
+                  </Radio>
+                </Stack>
+              </RadioGroup>
+            </VStack>
+          ),
+        )}
+      </VStack>
+      <Divider
+        pt="4"
+        pb="4"
+        borderColor="border.secondary.100"
+        textStyle="heading"
+      />
+      <Text textStyle="heading" pt="6">
+        By typing my name below, I hereby declare that I fully release and
+        discharge Focus on Nature, its employees and volunteers from all
+        liabilities to which I have agreed to above.
+      </Text>
+      <HStack pt="4" spacing="6">
```

Comment:
Can we add `flexWrap="wrap"` to this stack and used a fixed `vw` unit on the text boxes? I think the responsiveness makes more sense if it wraps, rather than just making the text boxes even less wide

Reply:
hm, I'm having some trouble doing this with `flexWrap="wrap"` on the `HStack`. I ended up finding and using `Wrap` component to do this.